cmd/kots/cli/deploy.go

```diff
@@ -0,0 +1,363 @@
+	}
+	newRequest.Header.Add("Authorization", authSlug)
+	newRequest.Header.Add("Content-Type", contentType)
+
+	resp, err := http.DefaultClient.Do(newRequest)
+	if err != nil {
+		return errors.Wrap(err, "failed to execute upstream update http request")
+	}
+	defer resp.Body.Close()
+
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return errors.Wrap(err, "failed to read upstream update server response")
+	}
+
+	response := struct {
+		Success bool   `json:"success"`
+		Error   string `json:"error,omitempty"`
+	}{}
+	_ = json.Unmarshal(respBody, &response)
+
+	if resp.StatusCode != http.StatusOK {
+		if response.Error != "" {
+			return errors.Errorf("upstream update failed: %s", response.Error)
+		}
+		return errors.Errorf("upstream update failed with status code %d", resp.StatusCode)
+	}
+
+	log.ActionWithoutSpinner("Upstream update processed successfully")
+
+	return nil
+}
+
+func handleAirgapImagePush(v *viper.Viper, clientset kubernetes.Interface, appSlug string, log *logger.CLILogger) error {
+	log.ActionWithoutSpinner("Pushing airgap images...")
+
+	airgapBundle := v.GetString("airgap-bundle")
+
+	registryConfig, err := getRegistryConfig(v, clientset, appSlug)
+	if err != nil {
+		return errors.Wrap(err, "failed to get registry config")
+	}
+
+	err = upstream.PushImagesFromAirgapBundle(airgapBundle, *registryConfig)
+	if err != nil {
+		return err
+	}
+
+	log.ActionWithoutSpinner("Images pushed successfully")
+	return nil
+}
+
+func handleSetConfigAndDeploy(v *viper.Viper, appSlug string, localPort int, authSlug string, log *logger.CLILogger) error {
+	log.ActionWithoutSpinner("Setting configuration and deploying...")
+
+	// Read config values from file (required flag)
+	configValues, err := os.ReadFile(v.GetString("config-values"))
+	if err != nil {
+		return errors.Wrap(err, "failed to read config values file")
+	}
+
+	requestPayload := map[string]interface{}{
+		"configValues":   configValues,
+		"deploy":         true, // HARDCODED - always deploy
+		"skipPreflights": v.GetBool("skip-preflights"),
+	}
+
+	requestBody, err := json.Marshal(requestPayload)
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal config request json")
+	}
+
+	url := fmt.Sprintf("http://localhost:%d/api/v1/app/%s/config/values", localPort, url.QueryEscape(appSlug))
+	newRequest, err := http.NewRequest("POST", url, bytes.NewBuffer(requestBody))
+	if err != nil {
+		return errors.Wrap(err, "failed to create config http request")
+	}
+	newRequest.Header.Add("Authorization", authSlug)
+	newRequest.Header.Add("Content-Type", "application/json")
+
+	resp, err := http.DefaultClient.Do(newRequest)
+	if err != nil {
+		return errors.Wrap(err, "failed to execute config http request")
+	}
+	defer resp.Body.Close()
+
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return errors.Wrap(err, "failed to read config server response")
+	}
+
+	response := struct {
+		Error string `json:"error"`
+	}{}
+	_ = json.Unmarshal(respBody, &response)
```

Comment:
can you handle this error always or at least for the successful case (if resp.StatusCode == http.StatusOK)